src/main/java/org/broadinstitute/hellbender/utils/smithwaterman/SmithWatermanAligner.java

```diff
@@ -0,0 +1,34 @@
+package org.broadinstitute.hellbender.utils.smithwaterman;
+
+import org.broadinstitute.gatk.nativebindings.smithwaterman.SWOverhangStrategy;
+import org.broadinstitute.gatk.nativebindings.smithwaterman.SWParameters;
+
+import java.io.Closeable;
+import java.util.function.Supplier;
+
+public interface SmithWatermanAligner extends Closeable{
+
+    SmithWatermanAlignment align(final byte[] ref, final byte[] alt, SWParameters parameters, SWOverhangStrategy overhangStrategy);
+
+    @Override
+    default void close() {}
```

Comment:
I don't believe it's generally safe to use after close. I'll update documentation.  I didn't want to require failure after close, because it's sort of a hassle to add to stateless implementations, how do you feel about "behavior of align after closing is undefined and is assumed to be unsafe."

Reply:
I think thats fine - it make it clear to consumers what to do without putting any burden on implementers like you say.